File: ios/Earwyrm/Core/Models/LyricNote.swift
import Foundation

struct LyricNote: Codable, Identifiable {
    let id: UUID
    let lyricId: UUID
    let userId: UUID
    let content: String
    let isPublic: Bool?
    let createdAt: Date
    let updatedAt: Date?

    enum CodingKeys: String, CodingKey {
        case id
        case lyricId = "lyric_id"
        case userId = "user_id"
        case content
        case isPublic = "is_public"
        case createdAt = "created_at"
        case updatedAt = "updated_at"
    }
}

struct NoteInsert: Encodable {
    let lyricId: UUID
    let content: String
    let isPublic: Bool

    enum CodingKeys: String, CodingKey {
        case lyricId = "lyric_id"
        case content
        case isPublic = "is_public"
    }
}

File: android/app/src/main/java/com/earwyrm/app/core/model/LyricNote.kt
package com.earwyrm.app.core.model

import kotlinx.datetime.Instant
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class LyricNote(val id: String, @SerialName("lyric_id") val lyricId: String, @SerialName("user_id") val userId: String, val content: String, @SerialName("is_public") val isPublic: Boolean? = null, @SerialName("created_at") val createdAt: Instant, @SerialName("updated_at") val updatedAt: Instant? = null)
@Serializable
data class LyricNoteUpsert(@SerialName("lyric_id") val lyricId: String, @SerialName("user_id") val userId: String, val content: String, @SerialName("is_public") val isPublic: Boolean = false)
